NEWS.rst

```diff
@@ -18,6 +18,8 @@ The following backend-specific members of the ``Dataset`` class have been remove
 They are replaced by backend-specific options in the JSON-based backend configuration.
 This can be passed in ``Dataset::options``.
 
+XY
+
```

Comment:
If breaking changes and major deprecations/upgrade guides are left, then they are to do here :)

Reply:
Let's add code snippets for a `shareRaw` to `load/storeChunk(Raw)` migration guide based on the diff in #1229.

We should show examples for `std::vector` and for raw pointers.